starboard/common/condition_variable.cc

```diff
@@ -66,13 +66,24 @@ bool ConditionVariable::WaitTimed(int64_t duration) const {
   bool was_signaled = SbConditionVariableIsSignaled(
       SbConditionVariableWaitTimed(&condition_, mutex_->mutex(), duration));
 #else
+  if (duration < 0) {
```

Comment:
This appears to address a symptom of a bug outside of this function.

If this ever is negative, then the caller is doing something wrong. Adding additional checks here makes every call to this function slower, and it is part of the critical path on some places where performance is critical. I would suggest making the fix outside of this function.